SmartDeviceLink/NSMutableDictionary+Store.m

```diff
@@ -24,30 +24,48 @@ - (nullable id)sdl_objectForName:(SDLName)name {
     return self[name];
 }
 
+- (nullable SDLEnum)sdl_enumForName:(SDLName)name {
+    return [self sdl_objectForName:name ofClass:NSString.class];
+}
+
+- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name {
+    return [self sdl_objectsForName:name ofClass:NSString.class];
+}
+
 - (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType {
     NSObject *obj = [self sdl_objectForName:name];
-    if (obj == nil || [obj isKindOfClass:classType]) {
+    if ([obj isKindOfClass:classType]) {
         return obj;
-    } else {
+    } else if ([obj isKindOfClass:NSDictionary.class] && [classType instancesRespondToSelector:@selector(initWithDictionary:)]) {
         return [[classType alloc] initWithDictionary:(NSDictionary *)obj];
+    } else if ([obj isKindOfClass:NSArray.class]) {
+        NSArray *array = (NSArray *)obj;
+        if ([array.firstObject isKindOfClass:classType]) {
+            return array.firstObject;
+        }
     }
+    return nil;
 }
 
-- (NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType {
-    NSArray *array = [self sdl_objectForName:name];
-    if ([array isEqual:[NSNull null]]) {
-        return [NSMutableArray array];
-    } else if (array.count < 1 || [array.firstObject isMemberOfClass:classType]) {
+- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType {
+    NSArray *array = [self sdl_objectForName:name ofClass:NSArray.class];
+    if (!array) {
+        return nil;
```

Comment:
What is the logic behind returning `nil` vs an empty array?

Reply:
empty array is broken current logic, you can see it if you change to [NSArray array] and run tests.